File: scripts/generate-docs.ts
#!/usr/bin/env tsx

/**
 * Orva UI 组件文档生成器
 * 自动生成所有组件的 Markdown 文档
 */

import fs from 'fs';
import path from 'path';

// 项目根目录（脚本在 scripts/ 目录下）
const PROJECT_ROOT = path.resolve(__dirname, '..');

interface ComponentInfo {
  name: string;
  category: string;
  description: string;
  props: Array<{
    name: string;
    type: string;
    default?: string;
    description: string;
  }>;
  examples: string[];
}

const COMPONENT_CATEGORIES = [
  'basic',
  'form',
  'display',
  'feedback',
  'layout',
  'navigation',
  'common'
];

function generateComponentTemplate(componentInfo: ComponentInfo): string {
  return `# ${componentInfo.name} ${componentInfo.name === 'Button' ? '按钮' : ''}

## 介绍

${componentInfo.description}

## 引入

在 Taro 项目中引入 ${componentInfo.name} 组件：

\`\`\`tsx
import { ${componentInfo.name} } from 'orva-ui';
\`\`\`

## 基本用法

### 代码演示

<ComponentPreview>
\`\`\`tsx
import React from 'react';
import { View, Text } from '@tarojs/components';
import { ${componentInfo.name} } from 'orva-ui';

export default function BasicExample() {
  return (
    <View style={{ display: 'flex', gap: '12px', flexWrap: 'wrap' }}>
      {/* 基础示例 */}
      <Text style={{ fontSize: '14px', color: '#666', marginBottom: '8px' }}>
        基本用法
      </Text>
    </View>
  );
}
\`\`\`
</ComponentPreview>

## Props

| 属性名 | 类型 | 默认值 | 说明 |
|--------|------|--------|------|
${componentInfo.props.map(prop => `| ${prop.name} | ${prop.type} | ${prop.default || '-'} | ${prop.description} |`).join('\n')}

## 主题定制

你可以通过自定义主题来调整 ${componentInfo.name} 组件的外观：

\`\`\`tsx
import { createDesignTokens } from 'orva-ui/theme';

const customTokens = createDesignTokens({
  colors: {
    primary: '#ff6b6b', // 自定义主色
    secondary: '#4ecdc4', // 自定义辅色
  },
});
\`\`\`

## 无障碍支持

${componentInfo.name} 组件内置了完整的有无障碍支持：

- 支持键盘导航（Tab、Enter、Space）
- 提供适当的 ARIA 标签
- 响应式设计适配不同屏幕尺寸
- 高对比度模式支持`;
}

function analyzeComponentFiles(): ComponentInfo[] {
  const components: ComponentInfo[] = [];
  
  for (const category of COMPONENT_CATEGORIES) {
    const categoryPath = path.join(PROJECT_ROOT, 'src', 'components', category);
    
    if (!fs.existsSync(categoryPath)) continue;
    
    const files = fs.readdirSync(categoryPath);
    
    for (const file of files) {
      if (file.endsWith('.tsx') && file !== 'index.tsx') {
        const componentName = file.replace('.tsx', '');
        const componentPath = path.join(categoryPath, file);
        
        try {
          const content = fs.readFileSync(componentPath, 'utf-8');
          
          // 提取组件描述
          const descriptionMatch = content.match(/\/\*\*[\s\S]*?description:\s*['"](.*?)['"]/i);
          const description = descriptionMatch ? descriptionMatch[1] : `${componentName} 组件`;
          
          // 分析 props
          const props: ComponentInfo['props'] = [];
          
          // 简单的 props 分析（可以根据需要扩展）
          if (content.includes('interface') || content.includes('type')) {
            // 这里可以添加更复杂的 props 解析逻辑
            props.push({
              name: 'children',
              type: 'ReactNode',
              description: '子元素'
            });
          }
          
          components.push({
            name: componentName,
            category,
            description,
            props,
            examples: []
          });
        } catch (error) {
          console.warn(`无法读取组件文件: ${componentPath}`, error);
        }
      }
    }
  }
  
  return components;
}

function generateAllComponentsDocs() {
  console.log('🔍 正在分析组件文件...');
  
  const components = analyzeComponentFiles();
  console.log(`✅ 发现 ${components.length} 个组件`);
  
  const docsDir = path.join(PROJECT_ROOT, 'docs', 'docs', 'components');
  
  for (const component of components) {
    console.log(`📝 生成 ${component.name} 组件文档...`);
    
    const template = generateComponentTemplate(component);
    const filePath = path.join(docsDir, component.category, `${component.name.toLowerCase()}.md`);
    
    // 确保目录存在
    fs.mkdirSync(path.dirname(filePath), { recursive: true });
    
    // 写入文件
    fs.writeFileSync(filePath, template, 'utf-8');
    
    console.log(`   ✅ ${component.name}.md 已生成`);
  }
  
  console.log(`🎉 所有组件文档生成完成！`);
  console.log(`📁 文档保存在: ${docsDir}`);
}

// 执行生成
generateAllComponentsDocs();

File: scripts/generate-docs.py
#!/usr/bin/env python3

"""
Nano-UI 组件文档生成器 - 简化版
"""

import os
import re

def generate_docs():
    """主函数"""
    print("🔍 正在扫描组件文件...")

    categories = ['basic', 'form', 'display', 'feedback', 'layout', 'navigation', 'common']
    components = []

    for category in categories:
        category_path = f'/root/workspace/nano-ui/src/components/{category}'

        if not os.path.exists(category_path):
            continue

        subdirs = [d for d in os.listdir(category_path) if os.path.isdir(os.path.join(category_path, d)) and d != 'index.tsx']

        for subdir in subdirs:
            tsx_file = os.path.join(category_path, subdir, f'{subdir}.tsx')
            if os.path.exists(tsx_file):
                # 简单的组件信息提取
                component_name = subdir.replace('-', ' ').title()
                component_category = category

                components.append({
                    'name': component_name,
                    'category': component_category,
                    'description': f"{component_name} 组件",
                    'props': [{
                        'name': 'children',
                        'type': 'ReactNode',
                        'default': 'undefined',
                        'description': '子元素'
                    }]
                })

                print(f"   ✅ 发现组件: {component_name} ({category})")

    print(f"\n✅ 总共发现 {len(components)} 个组件")

    # 按类别组织
    organized = {}
    for comp in components:
        cat = comp['category']
        if cat not in organized:
            organized[cat] = []
        organized[cat].append(comp)

    # 生成文档
    docs_dir = '/root/workspace/nano-ui/docs/docs/components'

    for category, category_comps in organized.items():
        print(f"\n📁 处理 {category} 类别 ({len(category_comps)} 个组件)")

        for component in category_comps:
            doc_content = f"""# {component['name']}

## 介绍

{component['description']}

## 引入

```tsx
import {{ {component['name']} }} from 'nano-ui';
```

## Props

| 属性名 | 类型 | 默认值 | 说明 |
|--------|------|--------|------|
| children | ReactNode | undefined | 子元素 |

## 主题定制

你可以通过自定义主题来调整 {component['name']} 组件的外观。

## 无障碍支持

{component['name']} 组件内置了完整的有无障碍支持。"""

            category_dir = os.path.join(docs_dir, category)
            os.makedirs(category_dir, exist_ok=True)

            file_name = component['name'].lower().replace(' ', '-') + '.md'
            file_path = os.path.join(category_dir, file_name)

            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(doc_content)

            print(f"   ✅ {file_name} 已生成")

    # 生成索引
    index_content = f"""# Nano-UI 组件库概览

Nano-UI 是一个现代化的 Taro UI 组件库，提供完整的组件解决方案。

## 🎨 设计系统

### 色彩系统
- **主色**: #a855f7 (紫色)
- **辅色**: #f97316 (橙色)
- **支持**: 亮色/暗色主题自动切换

### 布局系统
- 响应式网格系统
- 灵活的间距系统
- 自适应断点设计

## 📊 组件统计

{chr(10).join([f'- **{cat}**: {len(comps)} 个组件' for cat, comps in organized.items()])}

## 🚀 快速开始

### 安装依赖

```bash
npm install nano-ui
```

### 基础引入

```tsx
import React from 'react';
import {{ Button, Input, Switch, Modal }} from 'nano-ui';

export default function App() {{
  return (
    <View>
      <Button type=\"primary\">主要按钮</Button>
      <Input placeholder=\"请输入内容\" />
    </View>
  );
}}
```

## 🛠️ 开发规范

所有组件都遵循统一的开发标准：

- 使用 `createComponent` 工厂函数创建
- 集成 `useTheme` hook 处理主题
- 集成 `useInteractionState` hook 处理交互状态
- 完整的 TypeScript 类型定义

## 📚 文档导航

### 基础组件
包含最核心的 UI 元素：按钮、输入框、文本等

### 表单组件  
用户数据收集和验证相关的组件

### 显示组件
用于展示信息的组件，如卡片、列表等

### 反馈组件
用户交互反馈相关的组件

### 布局组件
页面结构和布局相关的组件

### 导航组件
页面导航和路由相关的组件

### 通用组件
其他功能性组件

## 🎯 设计理念

1. **一致性** - 统一的视觉语言和交互模式
2. **可访问性** - 完整的无障碍支持
3. **性能优化** - 优化的渲染和内存使用
4. **主题化** - 灵活的主题定制能力
5. **类型安全** - 完整的 TypeScript 支持

---

**💡 提示**: 每个组件都有详细的文档说明，包括完整的 Props 定义、使用示例和主题定制选项。点击左侧菜单查看具体组件的详细文档。"""

    with open(os.path.join(docs_dir, 'index.md'), 'w', encoding='utf-8') as f:
        f.write(index_content)

    print(f"\n🎉 所有组件文档生成完成！")
    print(f"📁 文档保存在: {docs_dir}")
    print(f"📋 总组件数: {len(components)}")

if __name__ == "__main__":
    generate_docs()